### ISSUE TITLE
Training the Unets

### ISSUE THREAD

**Issue by Shikamaru5** [lucidrains/imagen-pytorch#294]
So I want to be able to train the Unets in batches on many different datasets, and the LAION dataset which I will find easier to train in multiple steps. I believe that I may've found the way to do that but when I train the model and then load a new dataset and train it a second time, the PT file doesn't change in size. Is that normal or does that mean that I have done something wrong? I've included the current train file I'm using to train the model, incase there's something I'm missing. I do know that I'll need to train multiple Unets but I am wondering if perhaps I have to train a new Unet for every dataset, and then load them all in later when running the image generating task?

      from imagen_pytorch import Unet, Imagen, ImagenTrainer
      from imagen_pytorch.data import Dataset
      
      # unets for unconditional imagen
      
      unet = Unet(
          dim = 32,
          dim_mults = (1, 2, 4, 8),
          num_resnet_blocks = 1,
          layer_attns = (False, False, False, True),
          layer_cross_attns = False
      )
      
      # imagen, which contains the unet above
      
      imagen = Imagen(
          condition_on_text = False,  # this must be set to False for unconditional Imagen
          unets = unet,
          image_sizes = 64,
          timesteps = 1000,
      )
      
      trainer = ImagenTrainer(
          imagen,
          fp16 = True,
          split_valid_from_train = True,
          split_valid_fraction = 0.05,
      ).cuda()
      
      # instantiate your dataloader, which returns the necessary inputs to the DDPM as tuple in the order of images, text embeddings, then text masks. in this case, only images is returned as it is unconditional training
      
      train_ds = Dataset('3D_models', image_size = 128)
      
      trainer.add_train_dataset(train_ds, batch_size = 32)
      
      # working training loop
      
      for i in range(1):
          loss = trainer.train_step(unet_number = 1, max_batch_size = 8)
          print(f'loss: {loss}')
      
      trainer.save('./save.pt')
      
      trainer.load('./save.pt')

**Comment by Shikamaru5** [comment]
...Ya so, it turns out and this is something that is hardly written or talked about, probably should be. The model size doesn't change with more or less data. Of course a lot of you more veteran programmers likely knew that but geez that took me by surprise. With that I should be good to go then, and I'm super excited to see how it all works out! For anyone who has been reading my issues thank you for your patience with me.

**Comment by lijain** [comment]
Excuse me, do you have a super module behind the training? What is the effect?
